pycloudlib/ec2/instance.py

```diff
@@ -379,12 +466,61 @@ def _get_nic_matching_ip(self, ip_address):
             None,
         )
 
+    def _find_nic_associations(self, nics_resp: Dict) -> List[Dict]:
+        associations: List[Dict] = []
+        if not nics_resp["NetworkInterfaces"]:
+            return associations
+        for nic_md in nics_resp["NetworkInterfaces"]:
+            # Global associations
+            association = nic_md.get("Association")
+            if association:
+                associations.append(association)
+            # Per-ip associations
+            for priv_address in nic_md.get("PrivateIpAddresses", []):
+                if priv_address.get("Primary", False):
+                    continue
+                association = priv_address.get("Association")
+                if association:
+                    associations.append(association)
+        return associations
+
+    def _release_address(self, private_ip: str):
+        nics_resp = self._client.describe_network_interfaces(
+            Filters=[
+                {
+                    "Name": "private-ip-address",
+                    "Values": [private_ip],
+                },
+            ]
+        )
+        for association in self._find_nic_associations(nics_resp):
+            try:
+                self._client.disassociate_address(
+                    AssociationId=association["AssociationId"]
+                )
+                self._client.release_address(
```

Comment:
By placing these two operations in the same try block do we risk failing to attempt to release_address any time disassociate_address errors for each association? Is it worth separating these cases so we at least attempt both operations even if `disassociate_address` fails?

Reply:
If `disassocite_address` fails, it is very likely that `release_address` is going to fail, as in non-default VPCs, address must be disassociated as a precondition before `release_address` can be performed.

Reply:
Double try-except added, thanks.